tasks/gateway.yml

```diff
@@ -0,0 +1,79 @@
+---
+# TODO: `systemctl unmask tmp.mount` to make /tmp tmpfs (`cvmfs_server import`
+# fails if /tmp is xfs, although for some reason was fine on the PSU stratum 0
+# w/ xfs /tmp).
+
+- name: Determine whether -p flag is needed for cvmfs_server mkfs or import
+  ansible.builtin.set_fact:
+    cvmfs_config_apache_flag: "{{ '-p' if not cvmfs_config_apache else '' }}"
+
+- name: Include initial OS-specific tasks
+  ansible.builtin.include_tasks: init_{{ ansible_os_family | lower }}.yml
+  vars:
+    _cvmfs_role: gateway
+    _cvmfs_upgrade: "{{ cvmfs_upgrade_server }}"
+
+- name: Include key setup tasks
+  ansible.builtin.include_tasks: keys.yml
+
+- name: Include filesystem tasks
+  ansible.builtin.include_tasks: filesystem.yml
+
+- name: Include Apache tasks
+  ansible.builtin.include_tasks: apache.yml
+
+- name: Include firewall tasks
+  ansible.builtin.include_tasks: firewall.yml
+  vars:
+    _cvmfs_http_ports: "{{ cvmfs_gateway_http_ports }}"
+  when: cvmfs_manage_firewall
+
+- name: Include firewall tasks
+  ansible.builtin.include_tasks: firewall.yml
+  vars:
+    _cvmfs_gateway_ports: "{{ cvmfs_gateway_ports }}"
+  when: cvmfs_manage_firewall
+
+- name: Create repositories
+  ansible.builtin.command: |
+    /usr/bin/cvmfs_server mkfs {{ cvmfs_config_apache_flag }} -o {{ item.owner | default('root') }} -f {{ cvmfs_union_fs }} {{ item.repository }}
+  args:
+    creates: /srv/cvmfs/{{ item.repository }}
+  with_items: "{{ cvmfs_repositories }}"
+  notify:
+    - Restart apache
```

Comment:
I'm not sure to understand the gateway feature. Are not the repositories created on Stratum0? What kind of repositories does the role create here?